lib/printit/converter/html2pdf.rb

```diff
@@ -1,18 +1,36 @@
 module Printit
   class Html2Pdf
 
-    PRINCE_CONFIG = CONFIG_FILE.load['prince']
+    HANDLER = CONFIG_FILE.load['handler']
 
     def call(html)
+      case HANDLER
+      when "princexml"
+        convert_with_princexml(html)
+      when "weasyprint"
+        convert_with_weasyprint(html)
+      else
+        raise "Unsupported handler #{HANDLER}."
+      end
+    end
+
+  private
+
+    def convert_with_princexml(html)
       cmd = "prince #{prince_options} -"
       result, status = Open3.capture2(cmd, stdin_data: html)
       result
     end
 
-  private
-
     def prince_options
-      PRINCE_CONFIG.map{|k,v| "--#{k.gsub('_', '-')}=\"#{v}\"" }.join(" ")
+      config = CONFIG_FILE.load['prince']
+      config.map{|k,v| "--#{k.gsub('_', '-')}=\"#{v}\"" }.join(" ")
+    end
+
+    def convert_with_weasyprint(html)
+      cmd = "weasyprint -e utf-8 -v -f pdf - -"
+      result, status = Open3.capture2(cmd, stdin_data: html)
+      result
```

Comment:
Let see whether some weasyprint options could not be exposed in the printit.yml file, in a similar way to princexml.

Reply:
What options would you like to expose ?

Reply:
Oh I see, same options exist.